engine/execution/ingestion/engine.go

```diff
@@ -856,18 +878,19 @@ func newQueue(blockify queue.Blockify, queues *stdmap.QueuesBackdata) (*queue.Qu
 // A <- B <- C
 //   ^- D <- E <- F
 // Even through there are 6 blocks, we only need to check if block A becomes executable.
-// when the parent block isn't in the queue, we add it as a new queue. for instace, if
+// when the parent block isn't in the queue, we add it as a new queue. for instance, if
 // we receive H <- G, then the queues will become:
 // A <- B <- C
 //   ^- D <- E
 // G
-func enqueue(blockify queue.Blockify, queues *stdmap.QueuesBackdata) (*queue.Queue, bool) {
+func enqueue(blockify queue.Blockify, queues *stdmap.QueuesBackdata) (*queue.Queue, bool, bool) {
```

Comment:
Better to explain the two bool value. I think the first is `whether the block is a new block`, the second is `whether the block is the head of a queue`.

`queue, false, false`, means the block is a duplication
`queue, true, true`, means it's a block that is the head of a queue
`queue, true, false` means it's a block that is not the head of a queue, but extending a existing queue.`

Reply:
Yeah, definitely feel like once we get to two bools, might be more clear to either change to a "queue status" enum of some sort, or maybe a struct with two bool fields, for improved clarity.

Reply:
I don't think enum would help, as there are two decision to be made - first value (if fresh block/or not duplicate) let you bail out fast, second decides if execution can start. Representing this login with enum would seems more complicated.